src/Task/Git/CommitMessage.php

```diff
@@ -21,6 +22,11 @@ class CommitMessage implements TaskInterface
      */
     private $grumPHP;
 
+    /**
+     * @var array $exceptions
+     */
+    private $exceptions = [];
```

Comment:
Could you avoid the usage of this private property for temporary state inside the run context?

Reply:
No, I'dd rather keep it as a value object to feed the tasks. Allowing the context to be changed in between the tasks might cause a lot of issues.
I would avoid the local state since the run method can theoretically be called multiple times which causes an issue and we don't really need this state internally.
A try/catch over the if statement will solve the issue you are addressing as well.

Reply:
So if i am understanding correctly, that the task could run multiple times using a single instance which ***could*** persist the state of the `$exceptions` which would yield undesired behavior?

If this is accurate, couldn't we just reset exceptions back to an empty array in the `run()` method ?

Reply:
That is possible, but I prefer not to have it at all :)

Reply:
This should resolve this conversation https://github.com/phpro/grumphp/pull/541/commits/0056460816fd622c9986c5eb1dd8de82e29a467c